msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -177,9 +217,36 @@ An API-based message extension is created.
 
 :::image type="content" source="../assets/images/Copilot/api-based-me-tdp-plugin-copilot.png" alt-text="Screenshot shows the plugin for copilot app created in the app features page in Teams Developer Portal.":::
 
+**Add Authentication**
+
+To add authentication to your messag extesion, follow these steps:
+
+1. Go to **Tools** > **API Key Registration**.
+
+1. In the **Register an API key** page, update the following:
+
+   1. **Description**:
+   1. **Add Domain**:
+
+1. Under **Set a target tenant**, select the following:
+   1. If you want only the apps that are created within that tenant to use the API key to call the APIs, select **Home tenant**.
+   1. If you want only the apps that can within that tenant to use the API key to call the APIs, select **Any tenant**.
+
+1. Under **Set a Teams app**, select the following:
+   1. If you want the app that matches the given teams App ID to use the API key, select **Existing Teams app**.
+   1. If you want the app that matches the any teams App ID to use the API key, select **Any Teams app**.
+
+1. Select **+ Add Secret**.
```

Comment:
Please scrub all labels after we revamp the UI in TDP to make sure that they match. All these labels should be sentence case, and we will change the casing inside TDP.

Reply:
Updated the labels as per the latest updates in TDP.